packages/react-instantsearch/src/core/utils.js

```diff
@@ -58,3 +58,6 @@ export function assertFacetDefined(searchParameters, searchResults, facet) {
 export function getDisplayName(Component) {
   return Component.displayName || Component.name || 'UnknownComponent';
 }
+
+const resolved = typeof Promise !== 'undefined' && Promise.resolve();
```

Comment:
We can agree on supporting only browsers with Promise.resolve(), http://caniuse.com/#feat=promises

And delegate promise polyfill to the user.